client-js/app/elements/labrad-registry.ts

```diff
@@ -128,6 +128,69 @@ export class LabradRegistry extends polymer.Base {
       (reason) => self.handleError(reason)
     );
   }
+    
+  //Drag and drop
+  
+  //super annoying, but these have to be suppressed >:/
+  @listen('dragenter')
+  @listen('dragover')
+  suppressBehaviour(event) {
+    event.preventDefault();
+    console.log(event.target);
+    event.target.closest('tr').classList.add('over');
+  }
+
+   @listen('dragleave')
+   leaveDrag(event) {
+     event.preventDefault();
+     event.target.closest('tr').classList.remove('over');
+   }
+
+  @listen('dragstart')
+  startDrag(event) {
+   var data: any;
+   data = { path: this.path, name: event.target.name };
+   console.log('drag started', this.path);
+
+   event.dataTransfer.setData('text', JSON.stringify(data));
+   event.dataTransfer.effectAllowed = 'move'
+  }
+    
+  @listen('dragend')
+  endDrag(event) {
+    console.log('drag ended',event);
+  }
+
+  @listen('drop')
+  handleDrop(event) {
+
+    event.preventDefault();
+    var data: any;
+    data = JSON.parse(event.dataTransfer.getData('text'));
+
+    console.log('dropped on', event.target.closest('tr').name, 'with data', data ); 
+    if ( event.target.closest('tr').className.includes('dir') ) {
```

Comment:
style guide says no extra spaces in parentheses. Same thing in several places below.